cli/cli.go

```diff
@@ -0,0 +1,251 @@
+//	distributed under the License is distributed on an "AS IS" BASIS,
+//	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//	See the License for the specific language governing permissions and
+//	limitations under the License.
+package cli
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"runtime"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/pkg/errors"
+	"github.com/urfave/cli/v2"
+	terminal "golang.org/x/term"
+
+	snapshot_conf "github.com/mendersoftware/mender-snapshot/conf"
+
+	"github.com/mendersoftware/mender/conf"
+)
+
+const (
+	snapshotDescription = "Creates a snapshot of the currently running " +
+		"rootfs. The snapshots can be passed as a rootfs-image to the " +
+		"mender-artifact tool to create an update based on THIS " +
+		"device's rootfs. Refer to the list of COMMANDS to specify " +
+		"where to stream the image.\n" +
+		"\t NOTE: If the process gets killed (e.g. by SIGKILL) " +
+		"while a snapshot is in progress, the system may freeze - " +
+		"forcing you to manually hard-reboot the device. " +
+		"Use at your own risk - preferably on a device that " +
+		"is physically accessible."
+	snapshotDumpDescription = "Dump rootfs to standard out. Exits if " +
+		"output isn't redirected."
+)
+
+var (
+	errDumpTerminal = errors.New("Refusing to write to terminal")
+)
+
+type logOptionsType struct {
+	logLevel string
+	logFile  string
+	noSyslog bool
+}
+
+type runOptionsType struct {
+	config         string
+	fallbackConfig string
+	dataStore      string
+	keyPassphrase  string
+	bootstrapForce bool
+	conf.HttpConfig
+	logOptions logOptionsType
+}
+
+func ShowVersion() string {
+	return fmt.Sprintf("%s\truntime: %s",
+		snapshot_conf.VersionString(), runtime.Version())
+}
+
+func SetupCLI(args []string) error {
+	runOptions := &runOptionsType{}
+
+	app := &cli.App{
+		Name: "snapshot",
+		Usage: "Create filesystem snapshot -" +
+			"'mender-snapshot --help' for more.",
+		ArgsUsage:   "[options]",
+		Description: snapshotDescription,
+		Version:     ShowVersion(),
+		Commands: []*cli.Command{
+			{
+				Name:        "dump",
+				Description: snapshotDumpDescription,
+				Usage:       "Dumps rootfs to stdout.",
+				Action:      runOptions.DumpSnapshot,
+				Flags: []cli.Flag{
+					&cli.StringFlag{
+						Name: "source",
+						Usage: "Path to target " +
+							"filesystem " +
+							"file/directory/device" +
+							"to snapshot.",
+						Value: "/",
+					},
+					&cli.BoolFlag{
+						Name:    "quiet",
+						Aliases: []string{"q"},
+						Usage: "Suppress output " +
+							"and only report " +
+							"logs from " +
+							"ERROR level",
+					},
+					&cli.StringFlag{
+						Name:    "compression",
+						Aliases: []string{"C"},
+						Usage: "Compression type to use on the" +
+							"rootfs snapshot {none,gzip}",
+						Value: "none",
+					},
+				},
+			},
+		},
+	}
+	app.Flags = []cli.Flag{
+		&cli.StringFlag{
```

Comment:
We don't need the following flags:
```

   --fallback-config FILE, -b FILE  Fallback configuration FILE path. (default:
                                    "/var/lib/mender/mender.conf")
   --data DIR, -d DIR               Mender state data DIRECTORY path. (default:
                                    "/var/lib/mender")

   --trusted-certs FILE, -E FILE    Trusted server certificates FILE path.
   --forcebootstrap, -F             Force bootstrap. (default: false)

   --skipverify                     Skip certificate verification. (default: false)
   --passphrase-file value
```

Reply:
Looks to me like we only need the `log-level` flag actually